### PR TITLE
💄 181 - pdf download button state

### PR DESCRIPTION
**Pull request by anncatton**
Sets pdf download button text according to application state.
Fixes button appearance when in loading state.
Sets watermark to show on `REVISIONS REQUESTED` as well as `DRAFT` state
Hide Actions buttons when app is in `CLOSED` state

### CODE REVIEW

File: components/pages/Applications/ApplicationForm/Header/Actions.tsx [hunk 1]
```diff
@@ -24,10 +25,63 @@ import { getFormattedDate } from '../../Dashboard/Applications/InProgress/helper
...
-const HeaderActions = ({ appId }: { appId: string }): ReactElement => {
+// EXPIRED and RENEWING handling is tbd, CLOSED excludes Action buttons
+const getPdfButtonText = (state: ApplicationState) => {
+  let text = 'PDF';
+
+  if ([ApplicationState.DRAFT, ApplicationState.REVISIONS_REQUESTED].includes(state)) {
+    return (text = 'DRAFT ' + text);
+  }
+  if (isEqual(state, ApplicationState.SIGN_AND_SUBMIT)) {
+    return (text = 'FINALIZED ' + text);
```
Comment (ciaranschutte) [inline components/pages/Applications/ApplicationForm/Header/Actions.tsx:38]: might be a little cleaner to do: "return `Finalized ${text}`"

Comment (ciaranschutte) [reply components/pages/Applications/ApplicationForm/Header/Actions.tsx:38]: and changing the `text` to `const`

File: components/pages/Applications/ApplicationForm/Header/Actions.tsx [hunk 2]
```diff
@@ -24,10 +25,63 @@ import { getFormattedDate } from '../../Dashboard/Applications/InProgress/helper
...
+  }
+  if (
+    [ApplicationState.REVIEW, ApplicationState.APPROVED, ApplicationState.REJECTED].includes(state)
+  ) {
+    return (text = 'SIGNED ' + text);
+  }
+  console.warn('Illegal app state! State: ', state);
```
Comment (ciaranschutte) [inline components/pages/Applications/ApplicationForm/Header/Actions.tsx:45]: and return `text` if error. better than undefined.

**Review by ciaranschutte — APPROVED** [review]
one stylistic suggestion. looks good !

File: components/pages/Applications/ApplicationForm/Header/Actions.tsx (hunk 2 above)
Comment (anncatton) [reply components/pages/Applications/ApplicationForm/Header/Actions.tsx:45]: Yeah I was debating about what to do on this kind of error, but that sounds good for now.

**Pull request merged**